### PR TITLE
✅ Add visual and interactive test for amp-story-tooltip

### PR DESCRIPTION
**Pull request by Enriqe**
Part of #19181

### CODE REVIEW

File: examples/visual-tests/amp-story/amp-story-tooltip.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        font-weight: normal;
+        font-size: 20px;
+        text-transform: uppercase;
+        text-align: center;
+        color: white;
+      }
+    </style>
+    <script>
+      // TODO(#17108): Remove this when getMode().test is set.
+      AMP_TEST = true;
```
Comment (Enriqe) [inline examples/visual-tests/amp-story/amp-story-tooltip.html]: My tests wouldn't work unless this was set. (e.g. `await page.tap('a.title-small.center');` wouldn't actually show the tooltip in the story).

Linking #17108

@danielrozenberg

**Comment by gmajoulet** [comment]
We're using `justify-content: center;` in other places, I'd be surprised if it didn't work... This browser, even if old, should support flexboxes (esp since it even supports CSS grid).
I think it's worth investigating why it's broken on mobile, because it looks more like a height of the tooltip container problem: this could happen if the tooltip container's height is 0px, I believe

**Comment by Enriqe** [comment]
> because it looks more like a height of the tooltip container problem

But then the tooltip would also not be in the right position, since it's inside the container. 🤔

**Comment by newmuis** [comment]
The tooltip is absolutely positioned, right?  Then it wouldn't be affected, since it can be positioned outside of the container's bounds, even though it's a descendant.

**Comment by Enriqe** [comment]
After much debugging it looks like an older version of Firefox doesn't treat an absolute positioned element inside a flex-box same as newer versions.

I tweaked the CSS to make the nav arrows have vertical margins to make sure they work in other versions as well.

File: examples/visual-tests/amp-story/amp-story-tooltip-desktop.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * limitations under the License.
+ */
+'use strict';
+
+const {verifyCssElements} = require('../../../build-system/tasks/visual-diff/helpers');
+
+module.exports = {
+  'tapping on a clickable anchor should show the tooltip': async (page, name) => {
+    await page.tap('.next-container > button.i-amphtml-story-button-move');
+    await page.waitFor(1000);
```
Comment (danielrozenberg) [inline examples/visual-tests/amp-story/amp-story-tooltip-desktop.js]: Visual diff tests don't "really" run in parallel, they _load_ in parallel but are executed sequentially. If you can make this number smaller, it'll keep things running faster. This isn't a big deal on it's own (you're adding about 9 seconds of wait total in these two test cases) but these things add up :)

**Comment by gmajoulet** [comment]
General question as we start using visual diff testing, do we want to

(1) use it as integration testing, with tests such as "(tapping outside tooltip should hide it)"?
(2) use it to test the UI ("open the tooltip"), and keep the logic testing for our headless tests?

**Comment by danielrozenberg** [comment]
The line between unit and integration tests can be a bit blurry, especially in the case where both the described tests are testing the same "library". I would say that our visual tests should catch "interesting" behaviours - that is, they should ensure that the expected UI/UX doesn't break for users who are performing the same actions on components.

"do nothing" is a good test, "show tooltip" is a good test, and "hide tooltip" is a bad test _on its own_, but is a good test if there is also a "show tooltip" test

It's a difficult line to walk, because we don't have "multi-step" visual tests right now. Maybe I should add that as an option :)

/cc @rsimha if you have any additional comments

File: examples/visual-tests/amp-story/amp-story-tooltip.html (hunk 1 above)
Comment (danielrozenberg) [reply examples/visual-tests/amp-story/amp-story-tooltip.html]: Fixed in #19316, rebase on latest master if you need it!

File: examples/visual-tests/amp-story/amp-story-tooltip-desktop.js (hunk 2 above)
Comment (Enriqe) [reply examples/visual-tests/amp-story/amp-story-tooltip-desktop.js]: Added a commit that is smarter about when to use `waitFor` using ms.

File: examples/visual-tests/amp-story/amp-story-tooltip.html (hunk 1 above)
Comment (Enriqe) [reply examples/visual-tests/amp-story/amp-story-tooltip.html]: Ok that worked. Removed it from here.

**Comment by Enriqe** [comment]
Ready for another check @danielrozenberg

**Pull request merged**